apps/meteor/app/livechat/server/business-hour/Single.ts

```diff
@@ -26,6 +27,39 @@ export class SingleBusinessHourBehavior extends AbstractBusinessHourBehavior imp
 		return openBusinessHourDefault();
 	}
 
+	async onNewAgentCreated(agentId: string): Promise<void> {
+		businessHourLogger.debug('Executing onNewAgentCreated for agent in single BH mode', {
+			agentId,
+		});
+
+		const defaultBusinessHour = await LivechatBusinessHours.findOneDefaultBusinessHour();
+		if (!defaultBusinessHour) {
+			businessHourLogger.debug('No default business hour found for agentId', {
+				agentId,
+			});
+			return;
+		}
+
+		const businessHourToOpen = await filterBusinessHoursThatMustBeOpened([defaultBusinessHour]);
+		if (!businessHourToOpen.length) {
+			businessHourLogger.debug({
+				msg: 'No business hours found. Moving agent to NOT_AVAILABLE status',
+				agentId,
+				newStatus: ILivechatAgentStatus.NOT_AVAILABLE,
+			});
+			await Users.setLivechatStatus(agentId, ILivechatAgentStatus.NOT_AVAILABLE);
+			return;
+		}
+
+		await Users.addBusinessHourByAgentIds([agentId], defaultBusinessHour._id);
+
+		businessHourLogger.debug({
+			msg: 'Business hours found. Moving agent to AVAILABLE status',
+			agentId,
+			newStatus: ILivechatAgentStatus.AVAILABLE,
+		});
```

Comment:
are we going to debug every single step we code? is this really necessary ?

Reply:
I did found them quite useful while debugging locally but I guess I can re-check if we can remove some of them

Reply:
Removed a few logs. The once I kept around are at the end of an execution path, so although at first it might seem like a function has a bit too much logs, but keep in mind that at runtime only once of those paths will be executed so only one log will get emitted

Reply:
Side note: Yeah, we're doing some efforts on removing some of the logs, but on this specific feature, having a good amount of them is really useful to map problems.